Prefer unit file drop-ins to editing the unit file provided by the package manager

jsf9k opened the pull request:
## 🗣 Description ##

This pull request modifies the role to use drop-ins to alter the behavior of the `systemd` unit file, as opposed to editing the unit file provided by the package manager.

## 💭 Motivation and context ##

This pull request resolves #80.  Using a unit file drop-in is [the preferred method for making alterations to units](https://www.freedesktop.org/software/systemd/man/latest/systemd.unit.html#Examples).

## 🧪 Testing ##

All automated tests pass.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

jsf9k committed:
Prefer unit file drop-ins to editing the unit file provided by the package manager

Resolves #80.  Using a unit file drop-in is the preferred method in
the systemd documentation:
https://www.freedesktop.org/software/systemd/man/latest/systemd.unit.html#Examples

mcdonnnj commented on molecule/default/tests/test_default.py:
```diff
@@ -17,3 +18,20 @@ def test_sysctl_settings(host, setting):
+    "section,option,value",
+    [
+        ("Unit", "After", "network.target multi-user.target cloud-final.service"),
+        ("Service", "PrivateTmp", "False"),
```
Does it matter that the value you're looking for is `False` but the current value being set is `false`? I understand this value is likely a result of Python interpolation of the Ansible YAML so I just want to make sure that difference is not significant.

jsf9k replied:
It doesn't matter, for the reason you pointed out.

mcdonnnj replied:
Ah apologies that I was unclear. What I'm asking is will SystemD process a `False` the same as it would a `false`?

jsf9k replied:
The answer is yes, it would treat a `False` value the same as it would treat `false`, `off`, `Off`, `no`, or `No`.  I created commit 4371612, which I think makes it clearer what is going on.

It is unfortunate that the INI format isn't formally documented anywhere, but I think we can trust that systemd and Python's `configparser` library both interpret INI files in the expected and usual way.

jsf9k replied:
Digging deeper, I see [here](https://github.com/systemd/systemd/blob/main/src/shared/conf-parser.c#L999) that `systemd`'s config parser uses `parse_boolean()` to parse values that are expected to be Booleans.  That function is defined [here](https://github.com/systemd/systemd/blob/main/src/basic/parse-util.c#L22-L45), and one can see that a false value corresponds to a successful case-insensitive comparison against any of the strings `0`, `no`, `n`, `false`, `f`, or `off`.

mcdonnnj replied:
This works for me. My specific concern was that we previously wrote `PrivateTmp=false` as a string value and with Ansible processing we are now writing `PrivateTmp=False`. Since it appears that SystemD does a case-insensitive comparison with `false` (among others) then we are all good. Thank you for documenting the behavior for future reference.

mcdonnnj reviewed (approved):
LGTM ✔ One question for you though.

jsf9k committed:
Specify a type for the value of the unit file attribute when testing

Ansible roles use YML (which has a notion of types) and systemd unit
files use INI format (which does not and treats all values as
strings).  @mcdonnnj noted in
https://github.com/cisagov/ansible-role-openvpn/pull/81#discussion_r1653429157
that there was some confusion when I wrote a (Boolean) value of
"false" in tasks/main.yml but read back a (string) value of "False" in
the test code.  To make things more explicit, I now specify a type for
the systemd unit file attributes that I am testing.
configparser.ConfigParser supports interpretation of values as
strings, integers, floating-point numbers, or Booleans.

The reason for the discrepancy that @mcdonnnj observed is that:
1. Ansible's community.general.ini_file writes the YML value "false"
to the unit file as the string "False".
2. Python's configparser.ConfigParser.get() method interprets all
values as strings, and therefore returns the string "False".

Now what happens is:
1. Ansible's community.general.ini_file writes the YML value "false"
to the unit file as the string "False".
2. Python's configparser.ConfigParser.getboolean() method interprets all
values as Booleans, and therefore interprets the string "False" as a
false Boolean value.  See
https://docs.python.org/3/library/configparser.html#configparser.ConfigParser.getboolean
for more details.
